### PR TITLE
Remove tar.xz

### LINKED ISSUE #1576: Brand new realm 2.1.1 install, build fails (old lzma-native dependency)
## Goals

**What are you trying to achieve?**

Install realm in a react native project

**What did you expect to happen?**

Build successful

**What happened instead?**

Build fails. Unmet dependencies with **lzma-native**. 

It seems that when extracting _realm-sync-cocoa_ it looks for a _lzma-native_ module that doesn't exist (_binding-v3.0.4-node-v51-darwin-x64_ instead of _binding-v3.0.4-node-v59-darwin-x64_).

## Steps to Reproduce

Create a brand new react native project, install realm-js and build:

```
react-native init app
cd app
npm install realm --save
react-native link
```

Build fails.

## Code Sample

Xcode build output:

```
v7.10.0 is already installed.
Now using node v7.10.0 (npm v4.2.0)
Resolved requirements: { SYNC_SERVER_FOLDER: 'sync',
  SYNC_ARCHIVE: 'realm-sync-cocoa-2.1.10.tar.xz',
  SYNC_ARCHIVE_ROOT: 'core' }
Target directory has a differing lockfile, overwriting.
Extracting realm-sync-cocoa-2.1.10.tar.xz => /Users/ruben/dev/test/Wayne/node_modules/realm/vendor/realm-ios
{ Error: Cannot find module '/Users/ruben/dev/test/Wayne/node_modules/lzma-native/binding-v3.0.4-node-v51-darwin-x64/lzma_native.node'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.Module._load (module.js:418:25)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at /Users/ruben/dev/test/Wayne/node_modules/lzma-native/index.js:13:14
    at Object.<anonymous> (/Users/ruben/dev/test/Wayne/node_modules/lzma-native/index.js:595:3)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12) code: 'MODULE_NOT_FOUND' }
Command /bin/sh failed with exit code 1
```

Available binding folders for _lzma-native_:

```
binding-v3.0.1-node-v58-linux-x64
binding-v3.0.1-node-v59-linux-x64
binding-v3.0.3-node-v59-linux-x64
binding-v3.0.4-node-v59-darwin-x64
```

## Version of Realm and Tooling

- react: 16.0.0
- react-native 0.51.0
- realm 2.1.1
- lzma-native@3.0.4

### PR DESCRIPTION
**Pull request by blagoev**
fixes #1576 #1585

### CODE REVIEW

**Comment by blagoev** [comment]
this fails with 
ld: symbol(s) not found for architecture x86_64 for iOS build when migrating to the core parser

**Comment by kneth** [comment]
@blagoev I guess that neither Realm Core nor Realm Sync are released as `tar.gz`?

**Comment by blagoev** [comment]
@kneth The problem is in how sync is packaged. And this is already fix by https://github.com/realm/realm-sync/pull/1878

The build issues are related to the upgrade of object-store

File: realm.gypi [hunk 1]
```diff
@@ -175,7 +175,7 @@
         ]
       },
       "link_settings": {
-        "libraries": [ "-lrealm<(debug_library_suffix)" ],
+        "libraries": [ "-lrealm<(debug_library_suffix)", "-lrealm-parser<(debug_library_suffix)" ],
```
Comment (bdash) [inline realm.gypi]: These need to be linked in the reverse order so that GNU ld will find core symbols referenced by librealm-parser.a within librealm.a.

**Comment by blagoev** [comment]
Also the new query builder/argument converter does not throw when a bool column is filtered with invalid bool argument

**Comment by bdash** [comment]
I think the .xcodeproj needs to be updated for the various files that were removed + new libraries that need to be linked.

**Comment by bdash** [comment]
> Also the new query builder/argument converter does not throw when a bool column is filtered with invalid bool argument

If by "invalid bool argument" you mean a literal 0 or 1, then yes, that's expected. See [the discussion on Slack](https://realmio.slack.com/archives/C80PLGQ8Z/p1513380863000059).

**Comment by blagoev** [comment]
@bdash there is a test for "bool == 0" which expects an exception.

**Comment by blagoev** [comment]
@bdash so these two tests needs to be removed ?

**Comment by blagoev** [comment]
@bdash ignore I saw you change. I think I did not merge correctly into my branch

**Comment by blagoev** [comment]
The updated sync library does not include the required librealm-parser.a file. Once it is included in the sync release package this should build fine

**Comment by bdash** [comment]
That suggests this is blocked by https://github.com/realm/realm-sync/pull/1893.

**Pull request merged**